rules/apm/initial_access_cross_site_scripting.toml

```diff
@@ -0,0 +1,46 @@
+[metadata]
+creation_date = "2023/07/12"
+integration = ["apm"]
+maturity = "production"
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.3.0"
+updated_date = "2023/07/12"
+
+[rule]
+author = ["Elastic"]
+description = """
+Cross-Site Scripting (XSS) attacks are a type in which malicious scripts are injected into trusted websites.
+In XSS attacks, an attacker uses a benign web application to send malicious code, generally in the form of a
+browser-side script.This detection rule identifies the potential malicious executions of such browser-side scripts.
+"""
+index = ["apm-*-transaction*", "traces-apm*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Cross Site Scripting ( XSS )"
+references = ["https://github.com/payloadbox/xss-payload-list"]
+risk_score = 47
+rule_id = "4aa58ac6-4dc0-4d18-b713-f58bf8bd015c"
+severity = "medium"
+tags = ["Data Source: APM", "Use Case: Threat Detection", "Tactic: Initial Access"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+any where processor.name == "transaction" and
+url.fragment : ("*iframe*", "*prompt*", "*script*", "*svg*", "*onerror=*", "*javascript*alert*", "*eval*", "*onclick*",
```

Comment:
As is, `*script*` for example, will match unrelated stuff such as `description, subscription, etc`:
![image](https://github.com/elastic/detection-rules/assets/26856693/c24bab8a-697a-4510-9463-ff9dc63bb61e)

Reply:
I think there is a lot of room in matching page names too:
![image](https://github.com/elastic/detection-rules/assets/26856693/382e0248-8e50-4ddc-9076-66a48883a9c7)

Reply:
`*eval*` would match `retrieval` and potentially other unrelated stuff

Reply:
@shashank-elastic I think we can work on eliminating these partial matches before merging this. Consider combining them with other pieces of the payload.

Reply:
Have added move htm constructs like '<' , '=' in wildcard expresssions to mimic the payload.
With eval generally landing to a parenthesis payload those are additionally added aswell.